File: src-cpp/leetcode/remove-duplicates-from-sorted-array.cpp
#include <bits/stdc++.h>
using namespace std;

int removeDuplicates(vector<int> &nums)
{
    int n = nums.size();
    int i = 0;
    int k = 0;

    while (i < n)
    {
        auto num = nums[i];
        nums[k++] = num;

        while (i < n && nums[i] == num)
            i++;
    }

    return k;
}

File: src-ts/leetcode/remove-duplicates-from-sorted-array.ts
/* eslint-disable no-param-reassign */
function removeDuplicates(nums: number[]): number {
  if (nums.length <= 1) return nums.length;

  let k = 1;

  for (let i = 1; i < nums.length; i++) {
    if (nums[i] !== nums[i - 1]) {
      nums[k++] = nums[i];
    }
  }

  return k;
}
